controllers/upgrade/premisesChecker_test.go

```diff
@@ -0,0 +1,218 @@
+package upgrade
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/cloudogu/cesapp-lib/core"
+	k8sv1 "github.com/cloudogu/k8s-dogu-operator/api/v1"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_checkDoguIdentity(t *testing.T) {
+	t.Run("should succeed for dogus when forceUpgrade is off and remote dogu has a higher version", func(t *testing.T) {
+		localDogu := readTestDataLdapDogu(t)
+		remoteDogu := readTestDataLdapDogu(t)
+		remoteDogu.Version = "2.4.48-5"
+
+		// when
+		err := checkDoguIdentity(localDogu, remoteDogu, false)
+
+		// then
+		require.NoError(t, err)
+	})
+	t.Run("should succeed for dogus when forceUpgrade is on but would originally fail because of versions or names", func(t *testing.T) {
+		localDogu := readTestDataLdapDogu(t)
+		remoteDogu := readTestDataLdapDogu(t)
+		remoteDogu.Name = "different-ns/ldap"
+
+		// when
+		err := checkDoguIdentity(localDogu, remoteDogu, true)
+
+		// then
+		require.NoError(t, err)
+	})
+	t.Run("should fail for different dogu namespaces", func(t *testing.T) {
+		localDogu := readTestDataLdapDogu(t)
+		remoteDogu := readTestDataLdapDogu(t)
+		remoteDogu.Name = remoteDogu.GetNamespace() + "/test"
+		// when
+		err := checkDoguIdentity(localDogu, remoteDogu, false)
+
+		// then
+		require.Error(t, err)
+		assert.Equal(t, "dogus must have the same name (ldap=test)", err.Error())
```

Comment:
The name of the test is mistaken with next test and vice versa.

`"should fail for different dogu namespaces"` -> `"should fail for different dogu names"`
`"should fail for different dogu names"` -> `"should fail for different dogu namespaces"`

Reply:
- [ ] Done